addons/l10n_ar_withholding/wizards/account_payment_register.py

```diff
@@ -0,0 +1,199 @@
+            balance = self.company_currency_id.round(line.amount * conversion_rate)
+            payment_vals['write_off_line_vals'].append({
+                    **self._get_withholding_move_line_default_values(),
+                    'name': line.name,
+                    'account_id': account_id,
+                    'amount_currency': sign * line.amount,
+                    'balance': sign * balance,
+                    'tax_base_amount': sign * line.base_amount,
+                    'tax_repartition_line_id': tax_repartition_line_id,
+            })
+
+        for base_amount in list(set(self.withholding_ids.mapped('base_amount'))):
+            withholding_lines = self.withholding_ids.filtered(lambda x: x.base_amount == base_amount)
+            nice_base_label = ','.join(withholding_lines.mapped('name'))
+            account_id = self.company_id.l10n_ar_tax_base_account_id.id
+            base_amount = sign * base_amount
+            cc_base_amount = self.company_currency_id.round(base_amount * conversion_rate)
+            payment_vals['write_off_line_vals'].append({
+                **self._get_withholding_move_line_default_values(),
+                'name': _('Base Ret: ') + nice_base_label,
+                'tax_ids': [Command.set(withholding_lines.mapped('tax_id').ids)],
+                'account_id': account_id,
+                'balance': cc_base_amount,
+                'amount_currency': base_amount,
+            })
+            payment_vals['write_off_line_vals'].append({
+                **self._get_withholding_move_line_default_values(),  # Counterpart 0 operation
+                'name': _('Base Ret Cont: ') + nice_base_label,
+                'account_id': account_id,
+                'balance': -cc_base_amount,
+                'amount_currency': -base_amount,
+            })
+
+        return payment_vals
+
+    def _get_conversion_rate(self):
+        self.ensure_one()
+        if  self.currency_id !=  self.source_currency_id:
+            return self.env['res.currency']._get_conversion_rate(
+                self.currency_id,
+                self.source_currency_id,
+                self.company_id,
+                self.payment_date,
+            )
+        return  1.0
+
+    def _l10n_ar_get_payment_factor_dict(self):
+        # The factor represents the portion of the invoiced amount that I am paying and is used to calculate
+        # the base amount. When pay more than billed, the factor is 1, because I should only pay tax on the invoiced amount.
+        # billed | payment | factor
+        #    100 |      50 |    0.5
+        #    100 |     100 |      1
+        #    100 |     200 |      1
+
+        dict = {}
+        for rec in self:
+            amount_total = sum(rec.mapped('line_ids.move_id.amount_total'))
+            conversion_rate = rec._get_conversion_rate()
+            real_amount = min(rec.amount, rec.source_amount / conversion_rate)
```

Comment:
Yes It is in case when you pay more.
The rate is between the invoice currency and the wizard currency. 
I calculate it because is possible that neither the invoice nor the payment is in the company's currency.

Reply:
DONE!